commons/src/lib.rs

```diff
@@ -127,6 +144,26 @@ pub fn validate_content_type(
     }
 }
 
+/// get the requested version from accept header
+pub fn get_version(headers: &HeaderMap) -> i32 {
```

Comment:
It might be worth dropping `validate_content_type` and replacing it with a fallible `get_version`.  Because I don't see much use-case for a separate `validate_content_type` call.  Won't all the folks who call `validate_content_type` today call `get_version` right after?